### DISCUSSION TITLE
Kong all worker CPU 100%

### DISCUSSION THREAD

**Discussion by yuanst-627** [Kong/kong#13495] [Help]
Our production environment is using 100% CPU for all worker processes.

Export the flamegraph as follows
<img width="1296" alt="image" src="https://github.com/user-attachments/assets/a38fb97b-0281-4541-8f83-413c42f432c3">

![flame](https://github.com/user-attachments/assets/da55896b-aabb-47e4-b383-0dc988f25526)

./samples/lj-lua-stacks.sxx --skip-badvars -x 621952 > a.bt 
The results are as follows
[root@api-prd-api02 stapxx]# cat a.bt
0x7f17d4f3916a
T:@/usr/local/share/lua/5.1/kong/router.lua:1485
@/usr/local/share/lua/5.1/kong/router.lua:1485
@/usr/local/share/lua/5.1/kong/runloop/handler.lua:1091
@/usr/local/share/lua/5.1/kong/init.lua:752
=access_by_lua(nginx-kong.conf:257):0
        2118
0x7f17d4f39164
T:@/usr/local/share/lua/5.1/kong/router.lua:1485
@/usr/local/share/lua/5.1/kong/router.lua:1485
@/usr/local/share/lua/5.1/kong/runloop/handler.lua:1091
@/usr/local/share/lua/5.1/kong/init.lua:752
=access_by_lua(nginx-kong.conf:257):0
        391

**kong tag : 1.3.1**

Can someone help me out here? thanks

**Comment by ProBrian** [comment]
Hi @yuanst-627 , I noticed that you mentioned `kong tag: 1.3.1`, do you mean version here? If not, could you provide more information: what version of kong are your running? kong ce or kong ee? are you running a heavy workload? are there any logs with warning or error s, or is your kong service is not working functional correctly?

**Reply by yuanst-627** [reply]
<img width="1027" alt="image" src="https://github.com/user-attachments/assets/b046d311-6250-4939-8a1e-ae39141501d0">

I've observed this issue. Is there a possible connection?
https://github.com/openresty/luajit2/issues/42

**Comment by chronolaw** [comment]
I noticed this line "/usr/local/share/lua/5.1/kong/router.lua". Since kong gateway 3.x it will not exist.
I suggest to upgrade to latest version (3.7 or 3.4 LTS) and test again.
